src/rgw/rgw_rados.cc

```diff
@@ -1136,6 +1136,8 @@ int RGWRados::register_to_service_map(const string& daemon_type, const map<strin
   if (name.compare(0, 4, "rgw.") == 0) {
     name = name.substr(4);
   }
+  std::string instance_id = stringify(rados->get_instance_id());
+  std::string rgw_service_map_name = name + "." +  metadata["zonegroup_name"] + "." + metadata["zone_name"] + "." + instance_id;
```

Comment:
is this meant to be human-visible anywhere? or are we just adding this stuff to guarantee uniqueness? if the latter, i'd use "zone_id" and drop both zone_name and zonegroup_name

also, none of this accounts for different realms. but that might be out of scope?

Reply:
Yes, it's meant to be displayed as part of the ceph status so we want something human-readable. Yes, I wanted to add the realm but it does not seem to be populated anywhere at the moment so I left things the way they were. But if `svc` happens to provide the realm the rgw is part then we can certainly add it!

Reply:
@cbodley : any suggestion for the service_map_name.
1.) when name+realm_name+zone_name
2.) or just name+instance_id sufficient?
3.) adding zone id make it very difficult to understand IMO

Reply:
I added the realm too. Let's test again. Thanks

Reply:
Ok, looks like it's not building, if someone can have a look, I'm sure it's easy or someone familiar with the code :) Thanks!